consensus/core/src/leader_schedule.rs

```diff
@@ -851,16 +883,339 @@ mod tests {
 
         let mut dag_state_write = dag_state.write();
         dag_state_write.set_last_commit(last_commit);
-        dag_state_write.add_unscored_committed_subdags(unscored_subdags);
+        dag_state_write.add_scoring_subdags(unscored_subdags);
         drop(dag_state_write);
 
         assert_eq!(
             leader_schedule.elect_leader(4, 0),
             AuthorityIndex::new_for_test(0)
         );
 
-        leader_schedule.update_leader_schedule(&dag_state);
+        leader_schedule.update_leader_schedule_v2(&dag_state);
+
+        let leader_swap_table = leader_schedule.leader_swap_table.read();
+        assert_eq!(leader_swap_table.good_nodes.len(), 1);
+        assert_eq!(
+            leader_swap_table.good_nodes[0].0,
+            AuthorityIndex::new_for_test(2)
+        );
+        assert_eq!(leader_swap_table.bad_nodes.len(), 1);
+        assert!(
+            leader_swap_table
+                .bad_nodes
+                .contains_key(&AuthorityIndex::new_for_test(0))
+        );
+        assert_eq!(
+            leader_schedule.elect_leader(4, 0),
+            AuthorityIndex::new_for_test(2)
+        );
+    }
 
+    #[tokio::test]
+    async fn test_leader_schedule_from_store_with_vote_scoring() {
```

Comment:
Isn't the test `test_leader_schedule_from_store_with_no_scores_with_vote_scoring` missing?